// access-db/lib/constants/constants.js
"use strict";
/*
 * @Author: your name
 * @Date: 2020-05-18 18:02:23
 * @LastEditTime: 2020-06-06 08:45:37
 * @LastEditors: your name
 * @Description: In User Settings Edit
 * @FilePath: /@minappjs/weapp/src/constants/constants.ts
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.REDIS_STRUCTURE = exports.J_NAME_LIST = exports.J_MAX = exports.SET_METHORD = exports.FASTDB_UPDATE_METHORD = exports.REDIS_UPDATE_METHORD = exports.MYSQL_UPDATE_METHORD = exports.MONGODB_UPDATE_METHORD = exports.PLATFORM_NAME = void 0;
exports.PLATFORM_NAME = {
    MONGODB: 'mongodb',
    MYSQL: 'mysql',
    REDIS: 'redis',
    FASTDB: 'fastdb' //json本地 
};
//更新数据的方法
exports.MONGODB_UPDATE_METHORD = ['incr', 'set', 'unset', 'geo', 'append', 'remove', 'uAppend'];
exports.MYSQL_UPDATE_METHORD = ['incr', 'set', 'unset'];
exports.REDIS_UPDATE_METHORD = ['incr', 'add', 'set', 'unset', 'range', 'expireSec', 'expireMillisec', 'expireAt', 'expireMilliAt'];
exports.FASTDB_UPDATE_METHORD = ['incr', 'set', 'unset', 'geo', 'append', 'remove', 'uAppend'];
//新增数据方法
exports.SET_METHORD = [
    'geo'
];
exports.J_MAX = 10; // j*  参数的最大少于个数
exports.J_NAME_LIST = ['j0', 'j1', 'j2', 'j3', 'j4', 'j5', 'j6', 'j7', 'j8', 'j9'];
// redis 
exports.REDIS_STRUCTURE = {
    string: 'string'
};
